Add considerations about inert support (#471)
KittyGiraudel/a11y-dialog: docs/further_reading.migrating_to_v8.md

- It considers focusable the first `summary` element directly within a `details` element, provided it doesn’t have a negative tabindex attribute.
- It considers focusable `details` elements provided they do not have a `summary` element and do not have a negative tabindex attribute.

Finally, it now checks for the presence of the [`inert` attribute](https://developer.mozilla.org/en-US/docs/Web/API/HTMLElement/inert).

:::note  

> While the [support for `inert`](https://caniuse.com/mdn-api_htmlelement_inert)is increasing, it’s not perfect. The focus trap will ignore inert nodes and sub-trees **regardless** of the browser support for inert. If you want or need to rely on it, consider [using this polyfill](https://github.com/WICG/inert).  

Reviewer: ```suggestion
While the [support for `inert`](https://caniuse.com/mdn-api_htmlelement_inert)is increasing, it’s not perfect. The focus trap will ignore inert elements and all elements inside them, **regardless** of the browser's support for `inert`. If you want or need to rely on `inert`, consider [using this polyfill](https://github.com/WICG/inert).  
```

`node` feels a little jargon-y, but feel free to take or leave this suggestion :)
